gapis/replay/builder/builder.go

```diff
@@ -556,6 +556,28 @@ func (b *Builder) RegisterNotificationReader(reader NotificationReader) {
 	b.notificationReaders = append(b.notificationReaders, reader)
 }
 
+// Export compiles the replay instructions, returning a Payload that can be
+// sent to the replay virtual-machine.
+func (b *Builder) Export(ctx context.Context) (gapir.Payload, error) {
+	ctx = status.Start(ctx, "Export")
+	defer status.Finish(ctx)
+	ctx = log.Enter(ctx, "Export")
+
+	payload, _, _, err := b.Build(ctx)
+	if err != nil {
+		return payload, err
+	}
+
+	// Send phony postbacks.
+	byteOrder := b.memoryLayout.GetEndian()
+	for _, decoder := range b.decoders {
+		r := endian.Reader(bytes.NewReader([]byte{}), byteOrder)
```

Comment:
Nit: might be preferable to pretend there's at least the expected number of bytes returned. How about instead of `bytes.NewReader`, you use a dummy zero reader:

```go
type zeroReader struct{}

func (z zeroReader) Read(b []byte) (n int, err error) {
	for i := range b {
		b[i] = 0
	}
	return len(b), nil
}
```

Reply:
I'm thinking about just passing it an error that says the replay were never actually executed, it should handle that case gracefully anyway.

Reply:
SGTM